From c27f9215bde8942f4754e060cd769da672171c9b Mon Sep 17 00:00:00 2001
From: Ofir Fedida <ofedida@marvell.com>
Date: Tue, 24 May 2016 14:19:59 +0300
Subject: [PATCH 176/239] mv_ddr: apn806: Add static values to a80x0

Static values for DDR at 800Mhz for Dunit and Mckinley
memory controllers

Change-Id: I5c07ca30d7ef50af062e9d2aeb535597e9de6915
Signed-off-by: Ofir Fedida <ofedida@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29930
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>
---
 .../marvell/mv_ddr/apn806/mv_ddr_apn806_static.h   | 212 +++++++++++++++++++++
 1 file changed, 212 insertions(+)

diff --git a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h
index 6a6ba51..aa213c0 100644
--- a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h
+++ b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h
@@ -2576,10 +2576,222 @@ struct mk6_reg_data ddr_static_setup[] = {
 #elif defined(a80x0)
 #if defined(CONFIG_DUNIT_STATIC)
 struct mk6_reg_data ddr_static_setup[] = {
+	/* Traffic thru Tip::TIP init for Marvell with 1CS 8Gb x4 devices of Micron 2400 (17-17-17) */
+	{0xF0000030, 0x00000003}, /* set private ID*/
+	{0xF000070C, 0x000000FF}, /* enable SMC for STX access */
+	{0xF00116D8, 0x3CA},
+	{0xF06F0100, 0x4480000},
+	{0xF00119D4, 0x2},
+	{0xF06F8D36, 0xACAC0000},
+	{0xF06F4360, 0xFFFF0044},
+	{0xF00114C8, 0x5145941},
+	{0xF00117C8, 0x15145941},
+	{0xF0011DC8, 0x5145941},
+	{0xF0011EC8, 0x15145941},
+	{0xF00114CC, 0x1200D},
+	{0xF0011538, 0xB},
+	{0xF001153C, 0x13},
+	{0xf00116a0, 0xcc000000},	/* CKdelayADLL */
+	{0xf00116a0, 0xc0000017},	/* WLpup0 */
+	{0xf00116a0, 0xc0400012},	/* WLpup1 */
+	{0xf00116a0, 0xc080000d},	/* WLpup2 */
+	{0xf00116a0, 0xc0c00018},	/* WLpup3 */
+	{0xf00116a0, 0xc0020059},	/* RLpup0 */
+	{0xf00116a0, 0xc0420053},	/* RLpup1 */
+	{0xf00116a0, 0xc082004d},	/* RLpup2 */
+	{0xf00116a0, 0xc0c20084},	/* RLpup3 */
+	{0xf00116a0, 0xc0010021},	/* Ctxpup0 */
+	{0xf00116a0, 0xc041001c},	/* Ctxpup1 */
+	{0xf00116a0, 0xc0810017},	/* Ctxpup2 */
+	{0xf00116a0, 0xc0c1002a},	/* Ctxpup3 */
+	{0xf00116a0, 0xE8021503},	/* ADLL 0x92 */
+	{0xf00116a0, 0xE8130050},	/* ADLL */
+	{0xf00116a0, 0xE8140012},	/* ADLL */
+	{0xf00116a0, 0xEC021503},	/* ADLL */
+	{0xf00116a0, 0xEC130050},	/* ADLL */
+	{0xf00116a0, 0xEC140012},	/* ADLL */
+	{0xf00116a0, 0xF81B0020},	/* datareceivercalibrationBC */
+	{0xf00116a0, 0xE8280429},	/* vrefconfigurationBC */
+	{0xf00116a0, 0xEC24070C},	/* ctrldrivestrengthBC */
+	{0xf00116a0, 0xE824070E},	/* datadrivestrengthBC */
+	{0xf00116a0, 0xE82600C0},	/* dataODTBC */
+	{0xF0011400, 0x7b10cc30},
+	{0xF0011404, 0x36300848},
+	{0xF0011408, 0x5411cbb9},
+	{0xF001140c, 0x384A1f97},
+	{0xF0011410, 0x17114444},
+	{0xF0011414, 0x7c8},
+	{0xF0011424, 0x60f3f7},
+	{0xF001142c, 0x14c5138},
+	{0xF0011498, 0x3},
+	{0xF001149c, 0x301},
+	{0xF00114e0, 0x400},
+	{0xF00114e4, 0x3ff},
+	{0xF00114e8, 0x7f01ff},
+	{0xF00114ec, 0x54},
+	{0xF0011524, 0x8800},
+	{0xF00115e0, 0x1},
+	{0xF00115e4, 0x203c18},
+	{0xF00115ec, 0xd9ff0029},
+	{0xF0011900, 0x310},	/* MR0 */
+	{0xF0011904, 0x001},	/* MR1 */
+	{0xF0011908, 0x000},	/* MR2 */
+	{0xF001190c, 0x800},	/* MR3 */
+	{0xF0011910, 0x0},	/* MR4 */
+	{0xF0011914, 0x5A0},	/* MR5 */
+	{0xF0011918, 0x806},	/* MR6 */
+	{0xF0020020, 0x11000001},	/* MCinit */
+	{0xF0011480, 0x1},	/* init */
 	{-1, -1},
 };
 #else /* ddr static configuration thru mk6 */
 struct mk6_reg_data ddr_static_setup[] = {
+	/* Traffic thru MC6::TIP init for Marvell with 1CS 8Gb x4 devices of Micron 2400T */
+	{0xF00116D8, 0x3CC},
+	{0xF06F0100, 0x4480006},
+	{0xF00119D4, 0x2},
+	{0xF06F8D38, 0xACAC0000},
+	{0xF06F4360, 0xFFFF0044},
+	{0xF00114C8, 0x1840008},
+	{0xF00117C8, 0x28A0008},
+	{0xF0011DC8, 0x1840008},
+	{0xF0011EC8, 0x28A0008},
+	{0xf00116a0, 0xE83F0001},
+	{0xF00114CC, 0x1200D},
+	{0xF0011524, 0x8800},
+	{0xF0011400, 0x6b104c30},
+	{0xF0011400, 0x7b104c30},
+	{0xF0011538, 0xD0D},
+	{0xF001153C, 0x1515},
+
+	{0xF0020064, 0x606},		/* MC "readReady"+ MC2PHYlatency */
+	{0xF0021180, 0x500},
+	{0xF0021000, 0x60},		/* phy_rfifo_rptr_dly_val */
+	{0xF00210c0, 0x81000001},
+	{0xF00202c8, 0xfefe},
+
+	{0xf00116a0, 0xcc000000},	/* CKdelayADLL */
+	{0xf00116a0, 0xc0009c5f},
+	{0xf00116a0, 0xc0405c8f},
+	{0xf00116a0, 0xc0804c4b},
+	{0xf00116a0, 0xc0c02040},
+	{0xf00116a0, 0xc0020108},
+	{0xf00116a0, 0xc0420109},
+	{0xf00116a0, 0xc08200Cf},
+	{0xf00116a0, 0xc0c200de},
+	{0xf00116a0, 0xc0030007},
+	{0xf00116a0, 0xc043000A},
+	{0xf00116a0, 0xc0830008},
+	{0xf00116a0, 0xc0c30008},
+	{0xf00116a0, 0xc0040044},	/* WLpup0CS1 */
+	{0xf00116a0, 0xc044004E},	/* WLpup1 */
+	{0xf00116a0, 0xc0840010},	/* WLpup2 */
+	{0xf00116a0, 0xc0c4000B},	/* WLpup3 */
+	{0xf00116a0, 0xc006004B},	/* RLpup0 */
+	{0xf00116a0, 0xc046004C},	/* RLpup1 */
+	{0xf00116a0, 0xc0860015},	/* RLpup2 */
+	{0xf00116a0, 0xc0c60046},	/* RLpup3 */
+	{0xf00116a0, 0xc005000E},	/* Ctxpup0 */
+	{0xf00116a0, 0xc0450018},	/* Ctxpup1 */
+	{0xf00116a0, 0xc085001A},	/* Ctxpup2 */
+	{0xf00116a0, 0xc0c50015},	/* Ctxpup3 */
+	{0xf00116a0, 0xE8021503},	/* ADLL 0x92 */
+	{0xf00116a0, 0xE8130050},	/* ADLL */
+	{0xf00116a0, 0xE8140012},	/* ADLL */
+	{0xf00116a0, 0xEC021503},	/* ADLL */
+	{0xf00116a0, 0xEC130050},	/* ADLL */
+	{0xf00116a0, 0xEC140012},	/* ADLL */
+	{0xf00116a0, 0xF81B0020},	/* datareceivercalibrationBC */
+	/* {0xf00116a0, 0xE8280428},	 vrefconfigurationBC */
+	{0xf00116a0, 0xE8280427},	/* for debug  vref*/
+
+	{0xf00116a0, 0xEC24070C},	/* ctrldrivestrengthBC */
+
+	/* {0xf00116a0, 0xE824070E},	datadrivestrengthBC */
+	{0xf00116a0, 0xE8240489},
+
+	/* {0xf00116a0, 0xE8260040},	dataODTBC */
+	{0xf00116a0, 0xE82601C0},	/* for debug termination */
+
+	{0xF0011404, 0x5321BAA4},	/* 0x36301848 */
+	{0xF0011408, 0x5411cbb9},
+	{0xF001140c, 0x784A1f97},
+	{0xF0011410, 0x17114444},
+	{0xF0011414, 0x7c8},
+	{0xF0011424, 0x60f3f7},
+	{0xF001142c, 0x14c5138},
+	{0xF0011498, 0xf},
+	{0xF001149c, 0x303},
+	{0xF00114e0, 0x400},
+	{0xF00114e4, 0x3ff},
+	{0xF00114e8, 0x7f01ff},
+	{0xF00114ec, 0x54},
+	{0xF00115e0, 0x23},
+	{0xF00115e4, 0x203c18},
+	{0xF00115ec, 0xd9ff0029},
+	{0xF0011900, 0x314},	/* MR0 */
+	{0xF0011904, 0x001},	/* MR1 */
+	{0xF0011908, 0x200},	/* MR2 */
+	{0xF001190c, 0x800},	/* MR3 */
+	{0xF0011910, 0x0},	/* MR4 */
+	{0xF0011914, 0x4A0},	/* MR5 */
+	{0xF0011918, 0x80A},	/* MR6 */
+
+	{0xf0020340, 0x0},
+	{0xf0020344, 0x30000000},
+	{0xf0020310, 0x21000000},
+	{0xf0020314, 0x0},	/* TODO: for 2 cs add this configuration register */
+	{0xf0020318, 0x0},
+	{0xf002031c, 0x0},
+	{0xf0020304, 0x0},
+	{0xf0020308, 0x1},
+	{0xf0020200, 0x100001},
+	{0xf0020204, 0x0},
+	{0xf0020208, 0x0},
+	{0xf002020c, 0x0},
+	{0xf0020210, 0x0},
+	{0xf0020214, 0x0},
+	{0xf0020218, 0x0},
+	{0xf002021c, 0x0},
+	{0xf0020400, 0x100001},
+	{0xf0020404, 0x1},
+	{0xf0020408, 0x0},
+	{0xf002040c, 0x0},
+	{0xf0020220, 0x5010539},
+	{0xf0020224, 0x0},
+	{0xf0020228, 0x0},
+	{0xf002022c, 0x0},
+	{0xf0020044, 0x30300},
+	{0xf00202c0, 0x6000},
+	{0xf00202c4, 0x120030},
+	{0xf0020180, 0x30200},
+	{0xf0020050, 0xff},
+	{0xf002004c, 0x0},
+	{0xf0020054, 0x4c0},
+	{0xf0020300, 0x90b},
+	{0xf002030c, 0x90000},
+	{0xf0020380, 0x61a80},
+	{0xf0020384, 0x27100},
+	{0xf0020388, 0x960050},
+	{0xf002038c, 0x400},
+	{0xf0020390, 0x800200},
+	{0xf0020394, 0x1180618},
+	{0xf0020398, 0x1680300},
+	{0xf002039c, 0x200808},
+	{0xf00203a0, 0x1050600},
+	{0xf00203a4, 0x2},
+	{0xf00203a8, 0x1808},
+	{0xf00203ac, 0x11250b1B},	/*0x11260C1B RAS fix*/
+	{0xf00203b0, 0x0C0C060C},
+	{0xf00203b4, 0x06040602},
+	{0xf00203b8, 0x504},
+	{0xf00203bc, 0x1050505},
+	{0xf00203c0, 0x10400},		/*0x30700*/
+	{0xf00203c4, 0x0},
+	{0xf00203cc, 0xf010345},
+	/* {0xF06F8130, 0x1002A2A5},	AVS*/
+	{0xF0011480, 0x1},		Tipinit
+	{0xF0020020, 0x13000001},	MCinit
 	{-1, -1},
 };
 #endif
-- 
1.9.1

